clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaConsumerService.java

```diff
@@ -229,22 +231,34 @@ void unSubscribe(String versionTopic, String topic, int partition) {
   }
 
   void batchUnsubscribe(String versionTopic, Set<TopicPartition> topicPartitionsToUnSub) {
```

Comment:
Looking at the code, I see this `batchUnsubscribe` function being called only from the following function chain:

- `AggKafkaConsumerService::batchUnsubscribeConsumerFor`
- `SIT::consumerBatchUnsubscribe`
- `SIT::maybeUnsubscribeCompletedPartitions` (which short-circuits if hybrid or based on `server.unsub.after.batch.push`, which defaults to false)
- `SIT::processMessages` (which is a bit of a misnomer... this function used to process messages only in dedicated consumer mode, which we've deleted now, and now messages get processed from another code path)
- `SIT::run`

So anyway, the point I'm getting at is... while it may be good to improve the robustness of `batchUnsubscribe` here, I wonder how much does this actually help in practice, given that it's only called from fringe code path.

I see that you're adding a new call to this function from the test also, which replaces a call to `SharedKafkaConsumer::unSubscribe` but I wonder if it means that the call to `SKC::unSubscribe` remains fragile... there are many code paths calling `SKC::unSubscribe`, so are these paths safe? Should some or all of these code paths be modified also?

Reply:
In shared consumer, `unSubscribe` is synchronized and it will wait for the next poll to make sure all the pending messages are pushed to drainer queue.
`batchUnsubscribe` will be much efficient since it will only sync and wait once for many topic partition unsubscriptions, which I think it is still valuable to reduce the contention of the `SharedKafkaConsumer`.

Reply:
I think I was not clear. I think the change here is a good one, and I don't mean to say we should not do it. Rather, I am wondering if there are other places where we should be calling this function also, since currently it is being called from very few places, and and yet it might be better than the alternatives. But maybe there aren't other good places for calling this function, I am not sure.

Reply:
I think currently,  that is right place in StoreIngestionTask to unsubscribe all the completed partitions for batch-only stores since this unsubscription can't be executed for non-hybrid stores.

The `processMessages` is confusing because of the recent refactoring, and I am going to rename it to `checkIngestionProgress` to make it more clear.